### ISSUE TITLE
Using inversify with fastify-decorators

### ISSUE THREAD

**Issue by goriunov** [L2jLiga/fastify-decorators#762]
Hello,

I am trying to integrate inversify with the fastify-decorators, basically, all my services are registered through the inversify container and every time I try to inject them in to the controller I get "Invalid argument provided in XYZ constructor. Expected class annotated with @Service" 

after adding annotation to the service everything works but then if I try to inject some other deps in to the service I get the same issue.

Also if for any reason I have dynamic values injected (which are not classes) things become more messier, is there any way to make it all work together ?

Probably related to https://github.com/L2jLiga/fastify-decorators/issues/14

**Comment by L2jLiga** [comment]
Hi there, this is related to #14 and #395 both, in v4 it was done by moving injection stuff into separated library, you can try it if moving to pure ESM isn't problem for your project.

However I'm started to think about adding one more configuration option for bootstrap for both v3 and v4. 
The idea is to provide `classLoader` option which will be responsible for instantiating classes, by default for v3 it will be function that created classes with fastify-decorators DI mechanism, for v4 it's just about calling constructor

```typescript
import { fastify } from 'fastify'
import { bootstrap } from 'fastify-decorators'

const app = fastify()

app.register(bootstrap, {
  classLoader: (clazz) => Container.get(clazz)
})

**Comment by goriunov** [comment]
Thanks @L2jLiga for the answer, 

Yeah have been going through the code and trying out v4 as well but have been having troubles with moving to esm modules. We are considering to introduce fastify decorators to an existing set of projects and unfortunately, it is configured to compile to commonjs just adjusting some configs does not really work...

I have also tried to run examples from v4 branch and they also do not work by default when using npm, with the default v4 configurations provided no changes (node 16)...

```
 @fastify-decorators-examples/typedi@1.0.0 start
> node dist/start.js

node:internal/errors:465
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/XYZ/Desktop/projects/fastify-decorators/node_modules/fastify-decorators/index.js' imported from /Users/XYZ/Desktop/projects/fastify-decorators/examples/typedi/dist/index.js
    at new NodeError (node:internal/errors:372:5)
    at finalizeResolution (node:internal/modules/esm/resolve:437:11)
    at moduleResolve (node:internal/modules/esm/resolve:1009:10)
    at defaultResolve (node:internal/modules/esm/resolve:1218:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:580:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:294:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)
    at link (node:internal/modules/esm/module_job:78:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

That would be great if we had a version with common js which also allows custom DIs

**Comment by goriunov** [comment]
@L2jLiga  I think the solution with the 

```
app.register(bootstrap, {
  classLoader: (clazz) => Container.get(clazz)
})
```

should work nicely... We will need to handle the difference between Controllers registered through the simple DI and the other DI frameworks for injecting the rest of the dependencies.

**Comment by L2jLiga** [comment]
Released in [`v3.13.0`](https://github.com/L2jLiga/fastify-decorators/releases/tag/v3.13.0)  🎉 

I'm not sure how inversify works, but this hook should be enough to implement own logic for controllers creation
